src/lib/verifier/common.mli

```diff
@@ -0,0 +1,156 @@
+
+    Manually exposed as [bin_io_unversioned] cannot be used in signatures. *)
+
+val bin_size_invalid : invalid Bin_prot.Size.sizer
+
+val bin_write_invalid : invalid Bin_prot.Write.writer
+
+val bin_read_invalid : invalid Bin_prot.Read.reader
+
+val __bin_read_invalid__ : (int -> invalid) Bin_prot.Read.reader
+
+val bin_shape_invalid : Bin_prot.Shape.t
+
+val bin_invalid : invalid Bin_prot.Type_class.t
+
+(** Convert an [invalid] to a human-readable error. *)
+val invalid_to_error : invalid -> Error.t
+
+(** {2 Polymorphic verifiable command types}
+
+    These types abstract over the proof and auxiliary data types to support
+    both [User_command.Verifiable.t] (in-process) and
+    [User_command.Verifiable.Serializable.t] (cross-process via RPC). *)
+
+(** A zkApp command with verification key data attached.
+
+    {b Type parameters:}
+    - ['proof] is the proof type in account update authorizations
+    - ['aux] is auxiliary data attached to account updates *)
+type ('proof, 'aux) verifiable_zkapp_command =
+  ( ( Account_update.Body.t
+    , ('proof, Signature_lib.Schnorr.Chunked.Signature.t) Control.Poly.t
+    , 'aux )
+    Account_update.Poly.t
+  , Verification_key_wire.t option )
+  Zkapp_command.Call_forest.With_hashes_and_data.t
+  Zkapp_command.Poly.t
+
+(** A user command (signed or zkApp) in verifiable form.
+
+    {b Type parameters:}
+    - ['proof] is [Proof_cache_tag.t] for [User_command.Verifiable.t]
+      or [Pickles.Side_loaded.Proof.t] for [User_command.Verifiable.Serializable.t]
+    - ['aux] is [Account_update.T.Aux_data.t] for [User_command.Verifiable.t]
+      or [unit] for [User_command.Verifiable.Serializable.t] *)
+type ('proof, 'aux) verifiable_user_command =
+  ( Signed_command.t
+  , ('proof, 'aux) verifiable_zkapp_command )
+  User_command.Poly.t
+
+(** Verify a signed command's signature.
+
+    @return [Ok (`Assuming [])] if valid, [Error invalid] otherwise *)
+val check_signed_command :
+     signature_kind:Mina_signature_kind.t
+  -> Signed_command.t
+  -> ([ `Assuming of 'a list ], invalid) Result.t
+
+(** Collect verification key assumptions for zkApp proofs.
+
+    For each account update with proof authorization, collects the
+    (verification_key, statement, proof) tuple needed for async verification.
+
+    The return error type is open [[>]] to allow callers to extend with
+    additional error variants.
+
+    @return [Error invalid] if verification key is missing or mismatched *)
+val collect_vk_assumptions :
+     ('proof, _) verifiable_zkapp_command
+  -> ( (Side_loaded_verification_key.t * Zkapp_statement.t * 'proof) list
+     , [> `Missing_verification_key of
+          Signature_lib.Public_key.Compressed.t list
+       | `Unexpected_verification_key of
+         Signature_lib.Public_key.Compressed.t list ] )
+     Result.t
+
+(** Verify all signatures in a zkApp command.
+
+    {2 Verification Flow}
+
+    1. Compute transaction commitment from account updates
```

Comment:
This flow is too cliche and is basically 1-1 translation of the code. I'd just remove them.